mobilizon_reshare/main/publish.py

```diff
@@ -68,6 +69,11 @@ async def publish_event(
         return await publish_publications(publications)
 
 
+async def retrieve_publish(event_id, command_config: CommandConfig, publishers: Optional[Iterator[str]] = None):
+    event = await get_mobilizon_event_by_id(event_id)
+    return await publish_event(event, command_config, publishers)
```

Comment:
will this republish already published publications? or will just publish the unpublished ones? Because I believe it might do the first.

Reply:
imho this command is intended to be "destructive", if it's not scary enough i'm all for documenting it better. the idea is to have different layers of functionality, each one using capabilities from the layer below and offering capabilities to the layer above.

in my mind these are the layers (nothing cast in stone, i'm open to discuss this model). the higher the number, the higher the abstraction level, the lower the general-purposeness of the API:

1. `publish_event` and `publish_publications`: these functions are general purpose high-level interfaces to the `PublisherCoordinator`. the difference between the two of them is that `publish_event` will create new publications based on its `publishers` argument, or from settings if `None` is passed. They allow the administrator, in case of emergency, to override any selection logic and directly publish to platforms with `mobilizon-reshare publish -E $event_uuid` or `mobilizon-reshare publication retry $publication_uuid`
2. `select_and_publish` and `pull`: They allow for customization of pulling and publishing logic. An advanced user may think of pulling every 5 minutes to always have the latest updates from mobilizon and publishing with the configured strategy. `mobilizon-reshare pull` and `mobilizon-reshare publish` allow them to do that.
3. `start`: it's implemented in terms of `select_and_publish` and `pull` and allows a more casual user to schedule `mobilizon-reshare` default behavior.

To answer your question, this will create new publications based on it's `publishers` argument and try to publish them without looking at past failed or completed publications (there are no unpublished publications, only unpublished events. publications are created on-the-fly based on the publishers passed to `publish_event`)

Reply:
I believe documentation wouldn't do much. I suggest a dedicated flag like `--allow-republish` or something like that to indicate that this breaks something that the tool normally guarantees and forces the user to make an explicit choice to do it

Reply:
There is already one `--event` or `-E`. It may not be enough scary, so i changed it to `--force` or `-F` and updated the help message, it should be more explicit now.